pkg/analyze/node_resources.go

```diff
@@ -382,6 +394,26 @@ func nodeMatchesFilters(node corev1.Node, filters *troubleshootv1beta2.NodeResou
 		return true, nil
 	}
 
+	if filters.ResourceName != "" {
+		if filters.ResourceAllocatable != "" {
```

Comment:
```yaml
- nodeResources:
   filters:
     resourceName: gpu.intel.com/i915
```

Using the spec above, if a node without `gpu.intel.com/i915` is present, this code will match it and count it in as a node with the intel GPU

We need to add a check here that checks if `node.Status.Allocatable["gpu.intel.com/i915"]` or `node.Status.Capacity["gpu.intel.com/i915"]` exist

Reply:
Please add a unit test for this use case